app/src/main/java/com/google/homesampleapp/chip/ChipClient.kt

```diff
@@ -236,4 +236,65 @@ class ChipClient @Inject constructor(@ApplicationContext context: Context) {
     Timber.d("getDiscoveredDevice(${index})")
     return chipDeviceController.getDiscoveredDevice(index)
   }
+
+  suspend fun readAttribute(
+      devicePtr: Long,
+      attributePath: ChipAttributePath): AttributeState? {
+    return readAttributes(devicePtr, listOf(attributePath))[attributePath];
+  }
+
+  /**
+   * Wrapper around [ChipDeviceController.readAttributePath]
+   */
+  suspend fun readAttributes(
+      devicePtr: Long,
+      attributePaths: List<ChipAttributePath>): Map<ChipAttributePath, AttributeState> {
+    return suspendCoroutine { continuation ->
+        val callback: ReportCallback = object : ReportCallback {
+            override fun onError(
+                attributePath: ChipAttributePath?,
+                eventPath: ChipEventPath?,
+                e: Exception?
+            ) {
+                continuation.resumeWithException(IllegalStateException("readAttributes failed", e))
+            }
+
+            override fun onReport(nodeState: NodeState?) {
+                val states: HashMap<ChipAttributePath, AttributeState> = HashMap()
+                for(path in attributePaths) {
+                    var endpoint: Int = path.endpointId.id.toInt()
+                    states[path] = nodeState!!.getEndpointState(endpoint)!!
+                        .getClusterState(path.clusterId.id)!!
+                        .getAttributeState(path.attributeId.id)!!
+                }
+                continuation.resume(states);
+            }
+
+            override fun onDone() {
+                super.onDone()
+            }
+        }
+        chipDeviceController.readAttributePath(callback, devicePtr, attributePaths);
+      }
+  }
+
+  /**
+   * Wrapper around [ChipDeviceController.subscribeToAttributePath]
+   */
+  suspend fun subscribeToAttribute(
+      devicePtr: Long,
+      attributePath: ChipAttributePath,
+      minInterval: Int, maxInterval: Int,
+      callback: ReportCallback) {
+    return suspendCoroutine { continuation ->
+        chipDeviceController.subscribeToAttributePath(
+            { continuation.resume(Unit) },
+            callback,
+            devicePtr,
+            listOf(attributePath),
+            minInterval,
+            maxInterval
+        )
+    }
+  }
```

Comment:
Please run ktfmt to ensure standard formatting of all our source code.

Reply:
Done